src/MCPClient/lib/clientScripts/writeDigitalObjectComponentsToArchivesSpace.py

```diff
@@ -0,0 +1,80 @@
+
+
+def main(current_location, host, port, user, passwd):
+    dashboard_uuid = getDashboardUUID()
+
+    csvfile = open(os.path.join(current_location, 'metadata', 'digital_object_components.csv'), 'w')
+    writer = DictWriter(csvfile, ['component_id', 'digital_object_component_id'])
+    writer.writeheader()
+
+    component_ids = [os.path.basename(path).split('digital_object_component_')[1] for path in
+                     iglob(os.path.join(current_location, 'digital_object_component_*'))]
+
+    components = ArchivesSpaceDOComponent.objects.filter(id__in=component_ids)
+    # exit early to avoid the expense of hitting up ASpace
+    if components.count() == 0:
+        return 0
+
+    client = ArchivesSpaceClient(host=host, port=port, user=user, passwd=passwd)
+
+    # In most circumstances an AIP being processed will be associated
+    # with only a single archival object, but since that isn't guaranteed at
+    # the database level let's operate with the assumption this could happen.
+    archival_object_map = {}
+    for component in components:
+        if component.resourceid not in archival_object_map:
+            archival_object_map[component.resourceid] = []
+        archival_object_map[component.resourceid].append(component)
+
+    for resource_id, components in archival_object_map.iteritems():
+        do_id = client.add_digital_object(resource_id, dashboard_uuid, str(uuid4))['id']
+        for component in components:
+            if component.digitalobjectid:
+                continue  # already created
+
+            doc = client.add_digital_object_component(do_id,
+                                                      label=component.label,
+                                                      title=component.title)
+            # Track status about this component having been created for future use
+            component.digitalobjectid = do_id
+            component.remoteid = doc['id']
+            component.save()
+
+            writer.writerow({
+                'component_id': component.id,
+                'digital_object_component_id': component.remoteid,
+            })
+
+            print('Created digital object component "{}" associated with record "{}"'.format(doc['id'], resource_id))
+
+    return 0
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('--current-location', dest='current_location',
+                        type=str, help='%SIPDirectory%')
+    parser.add_argument('--host', default="localhost", dest="host",
+                        metavar="host", help="hostname of ArchivesSpace")
+    parser.add_argument('--port', type=int, default=8089, dest='port',
+                        metavar="port", help="Port used by ArchivesSpace backend API")
+    parser.add_argument('--user', dest='user', metavar="Administrative user")
+    parser.add_argument('--passwd', dest='passwd', metavar="Administrative user password")
```

Comment:
Tip: `dest` can be inferred from the first argument and the inferred one matches your specified one in every case. https://docs.python.org/2/library/argparse.html#dest